### DISCUSSION TITLE
Git LFS Upload even if Commit Rejected

### DISCUSSION THREAD

**Discussion by BenAigan** [git-lfs/git-lfs#4579] [Q&A] [answered]
We have various pre-commit hooks on our Bitbucket server to ensure conformity e.g. commit message validation, if these hooks prevent a commit which has LFS objects it will upload the LFS object successfully and then fail on the commit.

Is there a way to make this process atomic or is it this way by design?

**Comment by bk2204** [comment, marked as answer]
No, there isn't a way to make it atomic.  The LFS objects are uploaded first by the pre-push hook, and then the Git objects.  The pre-push hook is the only place that's possible to upload the objects to make them work properly with Git.

It's possible that a server implementation could quarantine LFS objects that are pushed and then remove them if the Git objects are rejected, but I'm not aware of any implementations that do that.
